frontend/app/components/shared/utils/query-param.ts

```diff
@@ -69,18 +69,24 @@ export const useQueryParam = () => {
         // get the start and end date option from the time range
         const { start: startOption, end: endOption } = getStartAndEndDate(timeRange);
 
-        start = isValidStartDate
-          ? startOption
-          : defaultDateOption?.startDate || lfxProjectDateOptions[1]?.startDate || null;
-
-        end = isValidEndDate
-          ? endOption
-          : defaultDateOption?.endDate || lfxProjectDateOptions[1]?.endDate || null;
+        if (timeRange === dateOptKeys.custom) {
+          start = paramStart as string;
+          end = paramEnd as string;
+        } else {
+          start = isValidStartDate
+            ? startOption
```

Comment:
Since its a common function i wouldnt stictly check certain parameters here, i would make a callback function to check for valid query params and pass it to the common function